File: apps/docs/src/code/python/agents/start_process.py
from os import getenv

from e2b import Sandbox

E2B_API_KEY = getenv("E2B_API_KEY")


def print_out(output):
    print(output.line)


def main():
    # 1. Start the playground sandbox
    sandbox = Sandbox(
        # Select the right runtime
        template="base",
        api_key=E2B_API_KEY,
    )

    # 2. Start the shell commdand
    proc = sandbox.process.start(  # $HighlightLine
        # Print names of all running processes
        cmd="ps aux | tr -s ' ' | cut -d ' ' -f 11",  # $HighlightLine
        on_stdout=print_out,  # $HighlightLine
        on_stderr=print_out,  # $HighlightLine
    )  # $HighlightLine

    # 3. Wait for the process to finish
    proc.wait()

    # 4. Or you can access output after the process has finished
    output = proc.output

    sandbox.close()


main()

File: apps/docs/src/code/js/agents/start_process.js
import { Session } from '@e2b/sdk'

// 1. Start the playground session
const session = await Session.create({
  // 'Node', 'Bash', 'Python3', 'Java', 'Go', 'Rust', 'PHP', 'Perl', 'DotNET'
  id: 'Nodejs',
  apiKey: process.env.E2B_API_KEY,
})

// 2. Start the shell commdand
let proc = await session.process.start({
  // $HighlightLine
  // Print names of all running processes
  cmd: "ps aux | tr -s ' ' | cut -d ' ' -f 11", // $HighlightLine
  // Stream stdout and stderr
  onStderr: data => console.log(data.line), // $HighlightLine
  onStdout: data => console.log(data.line), // $HighlightLine
}) // $HighlightLine

// 3. Wait for the process to finish
await proc.wait()

// 4. Or you can access output after the process has finished
const output = proc.output

await session.close()
